numpy/_core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -915,31 +997,81 @@ static GCC_CAST_OPT_LEVEL int
 #  if @is_complex2@
     dst_value[0] = _CONVERT_FN(src_value[0]);
     dst_value[1] = _CONVERT_FN(src_value[1]);
+    if (same_value_casting) {
```

Comment:
Should we not just have separate loops? (i.e., this would be `# if @same_value_casting@`). Or do you think (1) we can really count on the compiler always continuing to do the right thing, and (2) `-O3` will not ever screw up otherwise?

Reply:
In light of the performance drop, I think it makes sense to refactor this to create separate loops.